chain/actors/adt/diff/generic.go

```diff
@@ -79,7 +80,7 @@ type AdtMapDiff interface {
 	Remove(key string, val *typegen.Deferred) error
 }
 
-func DiffAdtMap(preMap, curMap Map, out AdtMapDiff) error {
+func GenericMap(preMap, curMap adt.Map, out AdtMapDiff) error {
```

Comment:
Maybe rename to `CompareMap`?

Reply:
Also rename `AdtMapDiff` to `MapComparer` or `MapDiffer`?

Reply:
done.